pkg/volume/util/atomic_writer.go

```diff
@@ -56,19 +56,24 @@ const (
 // inotify or fanotify to receive events when the content in the volume is
 // updated.
 type AtomicWriter struct {
-	targetDir  string
-	logContext string
+	targetDir   string
+	logContext  string
+	defaultMode os.FileMode
 }
 
 // NewAtomicWriter creates a new AtomicWriter configured to write to the given
 // target directory, or returns an error if the target directory does not exist.
-func NewAtomicWriter(targetDir, logContext string) (*AtomicWriter, error) {
+func NewAtomicWriter(targetDir string, defaultMode int32, logContext string) (*AtomicWriter, error) {
 	_, err := os.Stat(targetDir)
 	if os.IsNotExist(err) {
 		return nil, err
 	}
+	if defaultMode == 0 {
```

Comment:
correct.  The reason for demanding that optional fields be pointers is that we can distinguish between a user-provided 0 and not-specified.  In case of not-specified, set the value to the documented default.